### PR TITLE
polylith: Initial inclusion of polylith tooling

### PR DESCRIPTION
**Pull request by olymk2**
**Summary**

Inclusion of poly cmd tool, used to manage projects using the polylith architecture

**Test Plan**

Built installed the ran poly to may sure it entered the interactive interface.

**Checklist**

- [ ] Package was built and tested against unstable

### CODE REVIEW

File: packages/p/polylith/package.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+environment: |
+  export JAVA_CMD=/usr/lib64/openjdk-17/bin/java
+builddeps  :
+  - clojure-tools
+rundeps    :
+  - openjdk-17
+build      : |
+  clojure -T:build install :project poly
+install    : |
+  install -Dm00755 $pkgfiles/poly $installdir/usr/local/bin/poly
```
Comment (EbonJaeger) [inline packages/p/polylith/package.yml:22]: This should be going to `/usr/bin`. `/usr/local` is for programs that aren't managed by the system's package manager.

**Review by EbonJaeger — CHANGES_REQUESTED** [review]
The indentation for the `package.yml` should be 4 spaces instead of 2.

File: packages/p/polylith/package.yml (hunk 1 above)
Comment (olymk2) [reply packages/p/polylith/package.yml]: okay thanks for spotting that updated

**Review by EbonJaeger — CHANGES_REQUESTED** [review]
Thanks for fixing the install location. However, the indentation is still using the wrong number of spaces (2 instead of 4). :)

**Comment by olymk2** [comment]
> Thanks for fixing the install location. However, the indentation is still using the wrong number of spaces (2 instead of 4). :)

Bane of my life yaml indenting, not sure what rules my editor runs on save I have installed something called yaml-pro for next time to see if that helps, is there a cmd I can run that checks the file against the repositories requirements ?

**Comment by ReillyBrogan** [comment]
I don't see anything in this that will automatically use openjdk-17 if the $JAVA_HOME variable is unset. Do you know if if will do that automatically?

**Comment by olymk2** [comment]
There is this line exec "java" $JVM_OPTS "-jar" "/usr/lib/polylith/poly.jar" $ARGS its just calling the java binary you have enabled by default.

So a version of java needs to be available which is why its in the rundeps I believe they are moving to make this a compiled binary in the future so the jvm will not be required going forward.

Are you saying I need to set JAVA_HOME as it may not be set previously ?

**Comment by olymk2** [comment]
I have set the env in environment, using another package as a template so hopefully that's the issues resolved now, let me know if there is anything else

**Review by EbonJaeger — APPROVED** [review]
This looks good to me, but I'd like @ReillyBrogan to double-check with the Java stuff; I'm not as familiar with that as he is.

File: packages/p/polylith/files/poly [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#!/bin/sh
+
+ARGS=""
+while [ "$1" != "" ] ; do
+ ARGS="$ARGS $1"
+ shift
+done
+
+exec "java" $JVM_OPTS "-jar" "/usr/lib/polylith/poly.jar" $ARGS
```
Comment (ReillyBrogan) [inline packages/p/polylith/files/poly:9]: This is still going to have the same issue of not working if JAVA_HOME isn't set. Please take a look at the shim script located at `packages/d/davmail/files/davmail` to see how this is usually done. Basically, check to see if JAVA_HOME is defined, if not define it to the default and use it.

**Review by ReillyBrogan — CHANGES_REQUESTED** [review]
See comment

File: packages/p/polylith/files/poly (hunk 2 above)
Comment (olymk2) [reply packages/p/polylith/files/poly]: updated the shell script.

**Review by ReillyBrogan — APPROVED** [review]

**Pull request merged**